step.rb

```diff
@@ -0,0 +1,276 @@
+  return nil unless full_path
+  return nil unless File.exist? full_path
+  full_path
+end
+
+def export_dll(test_build_path)
+  dll_path = Dir[File.join(test_build_path, '/**/*.dll')].first
+  return nil unless dll_path
+
+  full_path = Pathname.new(dll_path).realpath.to_s
+  return nil unless full_path
+  return nil unless File.exist? full_path
+  full_path
+end
+
+def shutdown_simulators(xcode_major_version)
+  all_shotdown = all_simulator_shutteddown
+  return 0 if all_shotdown
+
+  `killall Simulator` if xcode_major_version == 7
+  `killall "iOS Simulator"` if xcode_major_version == 6
+  return 1 unless $?.success?
+
+  loop do
+    sleep 1 # second
+    all_shotdown = all_simulator_shutteddown
+    puts '    => waiting for shutdown ...'
+    break if all_shotdown
+  end
+  0
+end
+
+def boot_simulator(simulator, xcode_major_version)
+  simulator_cmd = ''
+  simulator_cmd = '/Applications/Xcode.app/Contents/Developer/Applications/Simulator.app' if xcode_major_version == 7
+  simulator_cmd = '/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/Applications/iPhone\ Simulator.app' if xcode_major_version == 6
+
+  `open #{simulator_cmd} --args -CurrentDeviceUDID #{simulator[:udid]}`
+  return 1 unless $?.success?
+
+  loop do
+    sleep 1 # seconds
+    out = `xcrun simctl openurl #{simulator[:udid]} https://www.google.com 2>&1`
+    puts '    => waiting for boot ...'
+    break if out == ''
+  end
+  sleep 2
+  0
+end
+
+def copy_app_to_simulator(simulator, app_path, xcode_major_version)
+  puts '  => shutdown simulators'
+  exit_code = shutdown_simulators(xcode_major_version)
+  return 1 if exit_code != 0
+
+  puts "  => erase simulator #{simulator}"
+  `xcrun simctl erase #{simulator[:udid]}`
+  return 1 unless $?.success?
+
+  puts "  => boot simulator #{simulator}"
+  exit_code = boot_simulator(simulator, xcode_major_version)
+  return 1 if exit_code != 0
+
+  puts "  => install .app #{app_path} to #{simulator}"
+  `xcrun simctl install #{simulator[:udid]} #{app_path}`
+  return 1 unless $?.success?
+  0
+end
+
+def run_unit_test(nunit_console_path, dll_path)
+  mono = '/Library/Frameworks/Mono.framework/Versions/Current/bin/mono'
+  out = `#{mono} #{nunit_console_path} #{dll_path}`
+  return [out, 1] unless $?.success?
+
+  regex = 'Tests run: (?<total>\d*), Errors: (?<errors>\d*), Failures: (?<failures>\d*), Inconclusive: (?<inconclusives>\d*), Time: (?<time>\S*) seconds\n  Not run: (?<not_run>\d*), Invalid: (?<invalid>\d*), Ignored: (?<ignored>\d*), Skipped: (?<skipped>\d*)'
+  match = out.match(regex)
+  unless match.nil?
+    _total, errors, failures, _inconclusives, _time, _not_run, _invalid, _ignored, _skipped = match.captures
+    return [out, 1] unless errors.to_i == 0 && failures.to_i == 0
+    return [match, 0]
+  end
+  [out, 1]
+end
+
+# -----------------------
+# --- main
+# -----------------------
+
+# Input validation
+xamarin_solution = ARGV[0]
```

Comment:
Use the same as here: https://github.com/gkiki90/steps-xamarin-archive/blob/master/step.rb#L35